app/services/set_localization_service.rb

```diff
@@ -0,0 +1,66 @@
+class SetLocalizationService
+  attr_reader :user, :http_accept_header
+  include Redmine::I18n
+
+  def initialize(user, http_accept_header)
+    @user = user
+    @http_accept_header = http_accept_header
+  end
+
+  ##
+  # Sets the locale.
+  # The locale is determined with the following priority:
+  #
+  #   1. The language as configured by the user.
+  #   2. The first language defined in the Accept-Language header sent by the browser.
+  #   3. OpenProject's default language defined in the settings.
+
+  def perform
```

Comment:
Regarding consistency:
I've seen services with a `call` method (nice, they behave like `Proc`s) and I've seen services with a `perform` method (nice, they behave like a `DelayedJob`.

I would prefer call, since a service should be unaware of its execution context and delayed jobs have some "know that you are a job" requirements...